src/tribler/core/components/ipv8/eva_protocol.py

```diff
@@ -144,32 +144,66 @@ class TransferResult:
     data: bytes
     nonce: int
 
+    def __str__(self):
+        return f'TransferResult(peer={self.peer}, info: {self.info}, data hash: {hash(self.data)}, nonce={self.nonce})'
+
+class TransferWindow:
+    def __init__(self, start: int, size: int):
+        self.data: List[Optional[bytes]] = [None for _ in range(size)]
+
+        self.size = size
+        self.start = start
+        self.processed: int = 0
+        self.last_window: bool = False
```

Comment:
**thought:** The `TransferWindow` itself does not use the `last_window` flag, so in my opinion, it is better to move it to `Transfer` because it is possible that in further simplifications, it may be removed entirely and replaced with something like `if len(transfer.data_list) == number_of_packets_to_receive: ...`

Reply:
I know. I placed it into the `Window` class just because of the feeling that this is safer rather than placing it into the `Transfer` class.
My explanation for this feeling is the following: A `Window` creates often. A `Transfer` creates only once. In case of any mistakes in the implementation of EVA, a more frequently created object (window) could level up some of the issues.

Reply:
Will be addressed here: https://github.com/Tribler/tribler/pull/6882